### PR TITLE
Build libcudf with -Wall

### PR DESCRIPTION
**Pull request by trxcllnt**
I discovered we're not building libcudf the `-Wall` GCC flag. This PR enables `-Wall` for GCC and nvcc, and fixes most of the errors.

~~The only error I haven't fixed yet is `-Werror=uninitialized` on this line [this line](https://github.com/rapidsai/cudf/blob/branch-0.18/cpp/include/cudf/scalar/scalar.hpp#L334), but @codereport is on it.~~ Fixed :heavy_check_mark:

### CODE REVIEW

**Review by codereport — APPROVED** [review]
:100:

**Comment by kkraus14** [comment]
Looks like `-Wall` isn't supported on `CUDA 10.1`?

File: cpp/benchmarks/common/generate_benchmark_input.cpp [hunk 1]
```diff
@@ -474,7 +474,7 @@ std::unique_ptr<cudf::column> create_random_column<cudf::list_view>(data_profile
     std::vector<int32_t> offsets{0};
     offsets.reserve(num_rows + 1);
     std::vector<cudf::bitmask_type> null_mask(null_mask_size(num_rows), ~0);
-    for (int row = 1; row < num_rows + 1; ++row) {
+    for (uint32_t row = 1; row < num_rows + 1; ++row) {
```
Comment (harrism) [inline cpp/benchmarks/common/generate_benchmark_input.cpp:477]:
```suggestion
    for (cudf::size_type row = 1; row < num_rows + 1; ++row) {
```

File: cpp/src/strings/regex/regcomp.cpp [hunk 2]
```diff
@@ -216,8 +216,8 @@ class regex_parser {
     }
 
     /* sort on span start */
-    for (int p = 0; p < cls.size(); p += 2)
-      for (int np = p + 2; np < cls.size(); np += 2)
+    for (size_t p = 0; p < cls.size(); p += 2)
```
Comment (harrism) [inline cpp/src/strings/regex/regcomp.cpp:219]: Should be `std::size_t` everywhere.

```suggestion
    for (std::size_t p = 0; p < cls.size(); p += 2)
```

**Review by harrism — CHANGES_REQUESTED** [review]
Use `std::size_t`. Also caught one case where `cudf::size_type` should be used.

File: cpp/benchmarks/common/generate_benchmark_input.cpp [hunk 3]
```diff
@@ -531,7 +531,7 @@ columns_vector create_random_columns(data_profile const& profile,
 std::vector<cudf::type_id> repeat_dtypes(std::vector<cudf::type_id> const& dtype_ids,
                                          cudf::size_type num_cols)
 {
-  if (dtype_ids.size() == num_cols) { return dtype_ids; }
+  if (dtype_ids.size() == static_cast<size_t>(num_cols)) { return dtype_ids; }
```
Comment (trxcllnt) [inline cpp/benchmarks/common/generate_benchmark_input.cpp:534]:
```suggestion
  if (dtype_ids.size() == static_cast<std::size_t>(num_cols)) { return dtype_ids; }
```

File: cpp/src/interop/to_arrow.cpp [hunk 4]
```diff
@@ -204,7 +204,7 @@ std::shared_ptr<arrow::Array> dispatch_to_arrow::operator()<cudf::struct_view>(
   arrow::MemoryPool* ar_mr,
   rmm::cuda_stream_view stream)
 {
-  CUDF_EXPECTS(metadata.children_meta.size() == input.num_children(),
+  CUDF_EXPECTS(metadata.children_meta.size() == static_cast<size_t>(input.num_children()),
```
Comment (trxcllnt) [inline cpp/src/interop/to_arrow.cpp:207]:
```suggestion
  CUDF_EXPECTS(metadata.children_meta.size() == static_cast<std::size_t>(input.num_children()),
```

File: cpp/src/interop/to_arrow.cpp [hunk 5]
```diff
@@ -300,13 +300,12 @@ std::shared_ptr<arrow::Table> to_arrow(table_view input,
                                        rmm::cuda_stream_view stream,
                                        arrow::MemoryPool* ar_mr)
 {
-  CUDF_EXPECTS((metadata.size() == input.num_columns()),
+  CUDF_EXPECTS((metadata.size() == static_cast<size_t>(input.num_columns())),
```
Comment (trxcllnt) [inline cpp/src/interop/to_arrow.cpp:303]:
```suggestion
  CUDF_EXPECTS((metadata.size() == static_cast<std::size_t>(input.num_columns())),
```

File: cpp/src/io/avro/avro.cpp [hunk 6]
```diff
@@ -135,7 +135,7 @@ bool container::parse(file_metadata *md, size_t max_num_rows, size_t first_row)
       if (parent_idx >= 0) {
         while (parent_idx >= 0) {
           if (md->schema[parent_idx].kind == type_union) {
-            int pos = parent_idx + 1;
+            size_t pos = parent_idx + 1;
```
Comment (trxcllnt) [inline cpp/src/io/avro/avro.cpp:138]:
```suggestion
            std::size_t pos = parent_idx + 1;
```

File: cpp/src/io/parquet/parquet.cpp [hunk 7]
```diff
@@ -285,7 +284,7 @@ bool CompactProtocolReader::read(KeyValue *k)
  */
 bool CompactProtocolReader::InitSchema(FileMetaData *md)
 {
-  if (WalkSchema(md) != md->schema.size()) return false;
+  if (static_cast<size_t>(WalkSchema(md)) != md->schema.size()) return false;
```
Comment (trxcllnt) [inline cpp/src/io/parquet/parquet.cpp:287]:
```suggestion
  if (static_cast<std::size_t>(WalkSchema(md)) != md->schema.size()) return false;
```

File: cpp/src/strings/regex/regcomp.cpp [hunk 8]
```diff
@@ -216,8 +216,8 @@ class regex_parser {
...
+      for (size_t np = p + 2; np < cls.size(); np += 2)
```
Comment (trxcllnt) [inline cpp/src/strings/regex/regcomp.cpp:220]:
```suggestion
      for (std::size_t np = p + 2; np < cls.size(); np += 2)
```

File: cpp/src/strings/regex/regcomp.cpp [hunk 9]
```diff
@@ -230,8 +230,8 @@ class regex_parser {
     /* merge spans */
     reclass yycls{builtins};
     if (cls.size() >= 2) {
-      int np = 0;
-      int p  = 0;
+      int np   = 0;
+      size_t p = 0;
```
Comment (trxcllnt) [inline cpp/src/strings/regex/regcomp.cpp:234]:
```suggestion
      std::size_t p = 0;
```

File: cpp/src/strings/regex/regcomp.cpp [hunk 10]
```diff
@@ -682,7 +682,7 @@ class regex_compiler {
     int rep_start = -1;
 
     out.clear();
-    for (int i = 0; i < in.size(); i++) {
+    for (size_t i = 0; i < in.size(); i++) {
```
Comment (trxcllnt) [inline cpp/src/strings/regex/regcomp.cpp:685]:
```suggestion
    for (std::size_t i = 0; i < in.size(); i++) {
```

File: cpp/src/strings/regex/regcomp.cpp [hunk 11]
```diff
@@ -701,11 +701,11 @@ class regex_compiler {
         regex_parser::Item item = in[i];
         if (item.d.yycount.n <= 0) {
           // need to erase
-          for (int j = 0; j < i - rep_start; j++) out.pop_back();
+          for (size_t j = 0; j < i - rep_start; j++) out.pop_back();
```
Comment (trxcllnt) [inline cpp/src/strings/regex/regcomp.cpp:704]:
```suggestion
          for (std::size_t j = 0; j < i - rep_start; j++) out.pop_back();
```

File: cpp/src/strings/regex/regcomp.cpp [hunk 12]
```diff
@@ -701,11 +701,11 @@ class regex_compiler {
...
         } else {
           // repeat
           for (int j = 1; j < item.d.yycount.n; j++)
-            for (int k = rep_start; k < i; k++) out.push_back(in[k]);
+            for (size_t k = rep_start; k < i; k++) out.push_back(in[k]);
```
Comment (trxcllnt) [inline cpp/src/strings/regex/regcomp.cpp:708]:
```suggestion
            for (std::size_t k = rep_start; k < i; k++) out.push_back(in[k]);
```

File: cpp/src/strings/regex/regcomp.cpp [hunk 13]
```diff
@@ -715,7 +715,7 @@ class regex_compiler {
             o_item.t    = LBRA_NC;
             o_item.d.yy = 0;
             out.push_back(o_item);
-            for (int k = rep_start; k < i; k++) out.push_back(in[k]);
+            for (size_t k = rep_start; k < i; k++) out.push_back(in[k]);
```
Comment (trxcllnt) [inline cpp/src/strings/regex/regcomp.cpp:718]:
```suggestion
            for (std::size_t k = rep_start; k < i; k++) out.push_back(in[k]);
```

File: cpp/src/strings/regex/regcomp.cpp [hunk 14]
```diff
@@ -746,7 +746,7 @@ class regex_compiler {
             }
           } else  // copy it once then put '*'
           {
-            for (int k = rep_start; k < i; k++) out.push_back(in[k]);
+            for (size_t k = rep_start; k < i; k++) out.push_back(in[k]);
```
Comment (trxcllnt) [inline cpp/src/strings/regex/regcomp.cpp:749]:
```suggestion
            for (std::size_t k = rep_start; k < i; k++) out.push_back(in[k]);
```

File: cpp/src/strings/regex/regcomp.cpp [hunk 15]
```diff
@@ -908,9 +908,9 @@ void reprog::optimize2()
 void reprog::print()
 {
   printf("Instructions:\n");
-  for (int i = 0; i < _insts.size(); i++) {
+  for (size_t i = 0; i < _insts.size(); i++) {
```
Comment (trxcllnt) [inline cpp/src/strings/regex/regcomp.cpp:911]:
```suggestion
  for (std::size_t i = 0; i < _insts.size(); i++) {
```

File: cpp/tests/copying/slice_tests.cpp [hunk 16]
```diff
@@ -273,7 +273,7 @@ TEST_F(SliceCornerCases, EmptyColumn)
   auto type_match_count = std::count_if(result.cbegin(), result.cend(), [](auto const& col) {
     return col.type().id() == cudf::type_id::EMPTY;
   });
-  EXPECT_EQ(type_match_count, expected);
+  EXPECT_EQ(static_cast<size_t>(type_match_count), expected);
```
Comment (trxcllnt) [inline cpp/tests/copying/slice_tests.cpp:276]:
```suggestion
  EXPECT_EQ(static_cast<std::size_t>(type_match_count), expected);
```

File: cpp/tests/copying/split_tests.cpp [hunk 17]
```diff
@@ -147,7 +147,7 @@ std::vector<cudf::table> create_expected_string_tables_for_splits(
   auto ret_cols_1 = create_expected_string_columns(strings[1], indices, validity[1]);
 
   std::vector<cudf::table> ret_tables;
-  for (int i = 0; i < ret_cols_0.size(); ++i) {
+  for (size_t i = 0; i < ret_cols_0.size(); ++i) {
```
Comment (trxcllnt) [inline cpp/tests/copying/split_tests.cpp:150]:
```suggestion
  for (std::size_t i = 0; i < ret_cols_0.size(); ++i) {
```

File: cpp/tests/copying/split_tests.cpp [hunk 18]
```diff
@@ -666,7 +666,7 @@ void split_null_input_strings_column_value(SplitFunc Split, CompareFunc Compare)
 
   auto expected = create_expected_string_tables_for_splits(strings, validity_masks, splits);
 
-  for (int i = 0; i < result.size(); ++i) { Compare(expected[i], result[i]); }
+  for (size_t i = 0; i < result.size(); ++i) { Compare(expected[i], result[i]); }
```
Comment (trxcllnt) [inline cpp/tests/copying/split_tests.cpp:669]:
```suggestion
  for (std::size_t i = 0; i < result.size(); ++i) { Compare(expected[i], result[i]); }
```

File: cpp/tests/copying/split_tests.cpp [hunk 19]
```diff
@@ -958,7 +958,7 @@ void split_nested_struct_of_list(SplitFunc Split, CompareFunc Compare)
   auto expected_struct_validity = create_expected_validity(splits, struct_validity);
   EXPECT_EQ(expected_names.size(), result.size());
 
-  for (int index = 0; index < result.size(); index++) {
+  for (size_t index = 0; index < result.size(); index++) {
```
Comment (trxcllnt) [inline cpp/tests/copying/split_tests.cpp:961]:
```suggestion
  for (std::size_t index = 0; index < result.size(); index++) {
```

File: cpp/tests/copying/split_tests.cpp [hunk 20]
```diff
@@ -1430,7 +1430,7 @@ TEST_F(ContiguousSplitNestedTypesTest, ListOfStruct)
   auto expected = cudf::split(static_cast<cudf::column_view>(*outer_list), splits);
   CUDF_EXPECTS(result.size() == expected.size(), "Split result size mismatch");
 
-  for (int index = 0; index < result.size(); index++) {
+  for (size_t index = 0; index < result.size(); index++) {
```
Comment (trxcllnt) [inline cpp/tests/copying/split_tests.cpp:1433]:
```suggestion
  for (std::size_t index = 0; index < result.size(); index++) {
```

File: cpp/tests/io/parquet_test.cpp [hunk 21]
```diff
@@ -2021,7 +2021,8 @@ TEST_F(ParquetReaderTest, DecimalRead)
       -9694494, -5343299, 3558393,  -8789072, 2697890,  -4454707, 8299309,  -6223703, -3112513,
       7537487,  825776,   -495683,  328299,   -4529727, 0,        -9999999, 9999999};
 
-    EXPECT_EQ(result.tbl->view().column(0).size(), sizeof(col0_data) / sizeof(col0_data[0]));
+    EXPECT_EQ(static_cast<size_t>(result.tbl->view().column(0).size()),
```
Comment (trxcllnt) [inline cpp/tests/io/parquet_test.cpp:2024]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(result.tbl->view().column(0).size()),
```

File: cpp/tests/io/parquet_test.cpp [hunk 22]
```diff
@@ -2045,7 +2046,8 @@ TEST_F(ParquetReaderTest, DecimalRead)
                            45844829680593,  71401416837149,  0,
                            -99999999999999, 99999999999999};
 
-    EXPECT_EQ(result.tbl->view().column(1).size(), sizeof(col1_data) / sizeof(col1_data[0]));
+    EXPECT_EQ(static_cast<size_t>(result.tbl->view().column(1).size()),
```
Comment (trxcllnt) [inline cpp/tests/io/parquet_test.cpp:2049]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(result.tbl->view().column(1).size()),
```

File: cpp/tests/io/parquet_test.cpp [hunk 23]
```diff
@@ -2160,7 +2162,8 @@ TEST_F(ParquetReaderTest, DecimalRead)
                            9913714, 901749,  7776938, 3186566, 4955569, 5131067, 98619,
                            2282579, 7521455, 4430706, 1937859, 4532040, 0};
 
-    EXPECT_EQ(result.tbl->view().column(0).size(), sizeof(col0_data) / sizeof(col0_data[0]));
+    EXPECT_EQ(static_cast<size_t>(result.tbl->view().column(0).size()),
```
Comment (trxcllnt) [inline cpp/tests/io/parquet_test.cpp:2165]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(result.tbl->view().column(0).size()),
```

File: cpp/tests/io/parquet_test.cpp [hunk 24]
```diff
@@ -2188,7 +2191,8 @@ TEST_F(ParquetReaderTest, DecimalRead)
                            0,
                            363993164092};
 
-    EXPECT_EQ(result.tbl->view().column(1).size(), sizeof(col1_data) / sizeof(col1_data[0]));
+    EXPECT_EQ(static_cast<size_t>(result.tbl->view().column(1).size()),
```
Comment (trxcllnt) [inline cpp/tests/io/parquet_test.cpp:2194]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(result.tbl->view().column(1).size()),
```

File: cpp/tests/partitioning/hash_partition_test.cpp [hunk 25]
```diff
@@ -70,7 +70,7 @@ TEST_F(HashPartition, ZeroPartitions)
   // Expect empty table with same number of columns and zero partitions
   EXPECT_EQ(input.num_columns(), output->num_columns());
   EXPECT_EQ(0, output->num_rows());
-  EXPECT_EQ(0, offsets.size());
+  EXPECT_EQ(size_t{0}, offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/hash_partition_test.cpp:73]:
```suggestion
  EXPECT_EQ(std::size_t{0}, offsets.size());
```

File: cpp/tests/partitioning/hash_partition_test.cpp [hunk 26]
```diff
@@ -90,7 +90,7 @@ TEST_F(HashPartition, ZeroRows)
...
```
Comment (trxcllnt) [inline cpp/tests/partitioning/hash_partition_test.cpp:93]:
```suggestion
  EXPECT_EQ(std::size_t{0}, offsets.size());
```

File: cpp/tests/partitioning/hash_partition_test.cpp [hunk 27]
```diff
@@ -107,7 +107,7 @@ TEST_F(HashPartition, ZeroColumns)
...
```
Comment (trxcllnt) [inline cpp/tests/partitioning/hash_partition_test.cpp:110]:
```suggestion
  EXPECT_EQ(std::size_t{0}, offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 28]
```diff
@@ -101,7 +101,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinPartitions13_3)
     }
 
     std::vector<cudf::size_type> expected_partition_offsets{0, 5, 9};
-    EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+    EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:104]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 29]
```diff
@@ -139,7 +139,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinPartitions13_3)
     }
 
     std::vector<cudf::size_type> expected_partition_offsets{0, 4, 9};
-    EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+    EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:142]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 30]
```diff
@@ -177,7 +177,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinPartitions13_3)
     }
 
     std::vector<cudf::size_type> expected_partition_offsets{0, 4, 8};
-    EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+    EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:180]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 31]
```diff
@@ -236,7 +236,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinPartitions11_3)
...
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:239]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 32]
```diff
@@ -273,7 +273,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinPartitions11_3)
     }
 
     std::vector<cudf::size_type> expected_partition_offsets{0, 3, 7};
-    EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+    EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:276]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 33]
```diff
@@ -310,7 +310,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinPartitions11_3)
     }
 
     std::vector<cudf::size_type> expected_partition_offsets{0, 4, 7};
-    EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+    EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:313]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 34]
```diff
@@ -370,7 +370,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinDegeneratePartitions11_15)
 
     std::vector<cudf::size_type> expected_partition_offsets{
       0, 0, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 11};
-    EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+    EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:373]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 35]
```diff
@@ -408,7 +408,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinDegeneratePartitions11_15)
 
     std::vector<cudf::size_type> expected_partition_offsets{
       0, 1, 2, 3, 4, 5, 6, 6, 6, 6, 6, 7, 8, 9, 10};
-    EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+    EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:411]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 36]
```diff
@@ -446,7 +446,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinDegeneratePartitions11_15)
 
     std::vector<cudf::size_type> expected_partition_offsets{
       0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10, 10, 10};
-    EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+    EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:449]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 37]
```diff
@@ -505,7 +505,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinDegeneratePartitions11_11)
     }
 
     std::vector<cudf::size_type> expected_partition_offsets{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
-    EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+    EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:508]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 38]
```diff
@@ -583,7 +583,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinNPartitionsDivideNRows)
     }
 
     std::vector<cudf::size_type> expected_partition_offsets{0, 7, 14};
-    EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+    EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:586]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 39]
```diff
@@ -630,7 +630,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinNPartitionsDivideNRows)
...
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:633]:
```suggestion
    EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/partitioning/round_robin_test.cpp [hunk 40]
```diff
@@ -687,7 +687,7 @@ TYPED_TEST(RoundRobinTest, RoundRobinSinglePartition)
   }
 
   std::vector<cudf::size_type> expected_partition_offsets{0};
-  EXPECT_EQ(num_partitions, expected_partition_offsets.size());
+  EXPECT_EQ(static_cast<size_t>(num_partitions), expected_partition_offsets.size());
```
Comment (trxcllnt) [inline cpp/tests/partitioning/round_robin_test.cpp:690]:
```suggestion
  EXPECT_EQ(static_cast<std::size_t>(num_partitions), expected_partition_offsets.size());
```

File: cpp/tests/strings/extract_tests.cpp [hunk 41]
```diff
@@ -140,7 +140,7 @@ TEST_F(StringsExtractTests, ExtractEventTest)
                                       "user@test.com",
                                       "Test Message Description"});
 
-  for (auto idx = 0; idx < patterns.size(); ++idx) {
+  for (size_t idx = 0; idx < patterns.size(); ++idx) {
```
Comment (trxcllnt) [inline cpp/tests/strings/extract_tests.cpp:143]:
```suggestion
  for (std::size_t idx = 0; idx < patterns.size(); ++idx) {
```

File: cpp/tests/utilities_tests/column_utilities_tests.cpp [hunk 42]
```diff
@@ -180,7 +180,7 @@ TEST_F(ColumnUtilitiesStringsTest, StringsToHostAllNulls)
     thrust::make_transform_iterator(h_strings.begin(), [](auto str) { return str != nullptr; }));
   auto host_data = cudf::test::to_host<std::string>(strings);
   auto results   = host_data.first;
-  EXPECT_EQ(3, host_data.first.size());
+  EXPECT_EQ(size_t{3}, host_data.first.size());
```
Comment (trxcllnt) [inline cpp/tests/utilities_tests/column_utilities_tests.cpp:183]:
```suggestion
  EXPECT_EQ(std::size_t{3}, host_data.first.size());
```

**Review by trxcllnt — COMMENTED** [review]
`size_t` -> `std::size_t`

**Review by kkraus14 — APPROVED** [review]
cmake approval

**Review by harrism — APPROVED** [review]

**Comment by kkraus14** [comment]
@trxcllnt Are we clear to remove the DO NOT MERGE label now and put on the auto merge?

**Comment by trxcllnt** [comment]
@kkraus14 yep

**Pull request merged**